src/fides/lib/oauth/api/routes/user_endpoints.py

```diff
@@ -248,11 +248,18 @@ def perform_login(
             client_secret_btye_length,
             scopes=user.permissions.scopes,  # type: ignore
             roles=user.permissions.roles,  # type: ignore
+            systems=user.system_ids,  # type: ignore
             user_id=user.id,
         )
-
-    if not user.permissions.scopes and not user.permissions.roles:  # type: ignore
-        logger.info("User needs scopes and/or roles to login.")
+    else:
+        # Refresh the client just in case
```

Comment:
Is there is a possibility of it not being accurate otherwise?

Reply:
I was thinking if roles or scopes were added directly through the database, or through some other location besides the one API endpoint that updates user permissions.

Reply:
I would hope users aren't doing that but I also appreciate you preempting unexpected behavior and adding a safety net, good thinking!